src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImpl.java

```diff
@@ -5,55 +5,159 @@
+        this.getControl().onMoveComplete();
     }
 
     @Override
-    public void onNodeMouseClick(NodeMouseClickEvent event) {
-        if (m_wiresManager.getSelectionManager() != null) {
-            m_wiresManager.getSelectionManager().selected(m_connector, event.isShiftKeyDown());
-        }
+    public void onNodeMouseClick(final NodeMouseClickEvent event) {
+       clickEventConsumer.accept(event);
     }
 
     @Override
-    public void onNodeMouseDoubleClick(NodeMouseDoubleClickEvent event) {
-        m_control.addControlPoint(event.getX(), event.getY());
+    public void onNodeMouseDoubleClick(final NodeMouseDoubleClickEvent event) {
+        doubleClickEventConsumer.accept(event);
     }
 
     public WiresConnectorControl getControl() {
-        return m_control;
+        return m_connector.getControl();
+    }
+
+    WiresConnector getConnector() {
+        return m_connector;
+    }
+
+    WiresManager getWiresManager() {
+        return m_wiresManager;
+    }
+
+    public static class DefaultClickHandlers {
+
+        private final Timer clickTimer;
+        private final WiresConnector connector;
+        private final boolean switchVisiblityOnClick;
+        private boolean isShiftKeyDown;
+
+        public DefaultClickHandlers(final WiresManager wiresManager,
+                                    final WiresConnector connector) {
+            this(connector,
+                 wiresManager.getSelectionManager(),
+                 true);
+        }
+
+        public DefaultClickHandlers(final WiresManager wiresManager,
+                                    final WiresConnector connector,
+                                    final boolean switchVisiblityOnClick) {
+            this(connector,
+                 wiresManager.getSelectionManager(),
+                 switchVisiblityOnClick);
+        }
+
+        private DefaultClickHandlers(final WiresConnector connector,
+                                    final SelectionManager selectionManager,
+                                     final boolean switchVisiblityOnClick) {
+            this.connector = connector;
+            this.switchVisiblityOnClick = switchVisiblityOnClick;
+            this.clickTimer = new Timer() {
+                @Override
+                public void run()
+                {
+                    if (selectionManager != null) {
+                        selectionManager.selected(connector, isShiftKeyDown);
+                    }
+                    if (switchVisiblityOnClick) {
+                        final WiresConnectorControl control = connector.getControl();
+                        if (control.areControlPointsVisible()) {
+                            control.hideControlPoints();
+                        } else {
+                            control.showControlPoints();
+                        }
+                    }
+                }
+            };
+        }
+
+        public Consumer<NodeMouseClickEvent> onClick() {
```

Comment:
maybe instead of `onClick` the method could have a name that represents the action that will be taken, and the parameters could be the one used by the actions in this case `isShiftKeyDown`